hail/src/main/scala/is/hail/annotations/BroadcastValue.scala

```diff
@@ -36,13 +36,18 @@ trait BroadcastRegionValue {
 
   def value: RegionValue
 
-  def t: PType
+  val t: PType
```

Comment:
Change to restrict mutable ptype, to allow (decodedPType, makeDec) to be lazy val, since buildDecoder is an expensive function. I don't think we should have a variable PType on a single instance of a BroadcastValue, but let me know if you disagree.

Alternative is I keep t `def` and use encoding.decodedPType(t.virtualType), then create a buildDecoded that take a physical type, and call that buildDecoder in the broadcast.